pathway-analysis/01-overrepresentation_analysis.Rmd

```diff
@@ -0,0 +1,327 @@
+
+```{r add_symbols, live = TRUE}
+neuron_markers_df <- neuron_marker_symbols %>%
+  # Using a left join removes the rows without gene symbols because those rows
+  # have already been removed in neuron_marker_symbols
+  dplyr::left_join(neuron_markers_df,
+                   # The name of the column that contains the Ensembl gene IDs
+                   # in the left data frame and right data frame
+                   by = c("ensembl_id" = "gene"))
+```
+
+## Over-representation Analysis (ORA)
+
+To test for over-representation, we can calculate a p-value with a hypergeometric test ([ref](https://yulab-smu.github.io/clusterProfiler-book/chapter2.html#over-representation-analysis)).
+
+\(p = 1 - \displaystyle\sum_{i = 0}^{k-1}\frac{{M \choose i}{{N-M} \choose {n-i}}} {{N \choose n}}\)
+
+Where `N` is the number of genes in the background distribution, `M` is the number of genes in a pathway, `n` is the number of genes we are interested in (our marker genes), and `k` is the number of genes that overlap between the pathway and our marker genes.
+
+Borrowing an example from [_clusterProfiler: universal enrichment tool for functional and comparative study_](https://yulab-smu.github.io/clusterProfiler-book/chapter2.html#fn2):
+
+> **Example**: Suppose we have 17,980 genes detected in a Microarray study and 57 genes were differentially expressed. Among the differential expressed genes, 28 are annotated to a gene set.
+
+We'll call genes that are differentially expressed `gene_in_interest` and genes that are in the gene set `in_gene_set`.
+
+```{r gene_table}
+gene_table <- data.frame(
+  gene_not_interest = c(2613, 15310),
+  gene_in_interest = c(28, 29)
+)
+rownames(gene_table) <- c("in_gene_set", "not_in_gene_set")
+
+gene_table
+```
+
+We can assess if the 28 overlapping genes mean that the differentially expressed genes are over-represented in the gene set with the hypergeometric distribution.
+This corresponds to a one-sided Fisher's exact test.
+
+```{r fisher_test}
+fisher.test(gene_table, alternative = "greater")
+```
+
+When we test **multiple pathways or gene sets**, the p-values then need to be **adjusted** for multiple hypothesis testing.
+
+### Neuron marker ORA
+
+Let's generate a list of genes that are indicative of neurons in the GBM data.
+We need to use the gene symbols, rather than the Ensembl gene IDs.
+
+```{r top_neuron_genes, live = TRUE}
+# Select genes that are in the top 100 for at least one comparison to other
+# cell types
+top_neuron_genes <- neuron_markers_df %>%
+  dplyr::filter(Top <= 100) %>%
+  dplyr::pull(gene_symbol)
+
+# Because genes can theoretically be listed as many times as there are other
+# cell types, we need to filter to unique genes
+top_neuron_genes <- unique(as.character(top_neuron_genes))
+```
+
+Maybe you're thinking, "Hey, picking the top 100 genes seems kind of arbitrary."
```

Comment:
There are actually 415 genes in this set, as it is the union of many top 100 lists. This might be confusing?

Reply:
This seems like a verbal point to me, or at least I'm not sure if writing this is "the union of many top 100 lists" will be helpful or more confusing?

Reply:
Verbal is fine. maybe just
```suggestion
Maybe you're thinking, "Hey, picking the top 100 genes for each comparison seems kind of arbitrary."
```

Reply:
Yes, this sounds good, thanks!